thesslstore_module.php

```diff
@@ -66,6 +66,276 @@ public function getAuthors() {
         return self::$authors;
     }
 
+    /**
+     * Performs any necessary bootstraping actions
+     */
+    public function install()
+    {
+        // Add cron tasks for this module
+        $this->addCronTasks($this->getCronTasks());
+    }
+
+    /**
+     * Performs any necessary cleanup actions
+     *
+     * @param int $module_id The ID of the module being uninstalled
+     * @param boolean $last_instance True if $module_id is the last instance across
+     *  all companies for this module, false otherwise
+     */
+    public function uninstall($module_id, $last_instance)
+    {
+        if (!isset($this->Record)) {
+            Loader::loadComponents($this, ['Record']);
+        }
+        Loader::loadModels($this,  ['CronTasks']);
+
+        $cron_tasks = $this->getCronTasks();
+
+        if ($last_instance) {
+            // Remove the cron tasks
+            foreach ($cron_tasks as $task) {
+                $cron_task = $this->CronTasks->getByKey($task['key'], $task['dir'], $task['task_type']);
+                if ($cron_task) {
+                    $this->CronTasks->deleteTask($cron_task->id, 'module', $task['dir']);
```

Comment:
How come you use the `$task['task_type']` above, but then use the string 'module' for the task type here?